zipline/pipeline/data/dataset.py

```diff
@@ -105,7 +105,7 @@ def __new__(mcls, name, bases, dict_):
                 setattr(newtype, maybe_colname, bound_column)
                 _columns.append(bound_column)
 
-        newtype._columns = _columns
+        newtype._columns = frozenset(_columns)
```

Comment:
Wondering if we should make this a tuple ordered by column name rather than a `frozenset`.  Having consistent ordering on DataSet.columns seems nice.  Thoughts?